java/java.hints/src/org/netbeans/modules/java/hints/jdk/ConvertToVT.java

```diff
@@ -0,0 +1,132 @@
+import org.netbeans.spi.java.hints.JavaFix;
+import org.netbeans.spi.java.hints.TriggerPattern;
+import org.netbeans.spi.java.hints.TriggerPatterns;
+import org.openide.util.NbBundle;
+
+/**
+ *
+ * @author mjayan
+ */
+@NbBundle.Messages({
+    "DN_ConvertToVT=Convert to Virtual Thread Executor",
+    "DESC_ConvertToVT=Convert to Virtual Thread Executor",
+    "ERR_ConvertToVT=Convert to Virtual Thread Executor",
+    "FIX_ConvertToVT=Convert to Virtual Thread Executor",
+    "ERR_ConvertToThreadPerTask=Convert to newThreadPerTask executor"
+})
+@Hint(displayName = "#DN_ConvertToVT", description = "#DESC_ConvertToVT", category = "rules15",
+        minSourceVersion = "19")
+public class ConvertToVT {
+
+    private static final int VT_PREVIEW_JDK_VERSION = 19;
+
+    @TriggerPatterns({
+        @TriggerPattern(value = "$var = $expr2.$meth1($var1$);"),
+        @TriggerPattern(value = "$type $var = $expr2.$meth1($var1$);")
+    })
+    public static ErrorDescription compute(HintContext ctx) {
+        if (Utilities.isJDKVersionLower(VT_PREVIEW_JDK_VERSION) && !CompilerOptionsQuery.getOptions(ctx.getInfo().getFileObject()).getArguments().contains("--enable-preview")) {
+            return null;
+        }
+        List<String> methList = Arrays.asList("newFixedThreadPool", "newCachedThreadPool");
+        String expr = ctx.getInfo().getTrees().getElement(ctx.getVariables().get("$expr2")).getSimpleName().toString();
+        String method = ctx.getVariableNames().get("$meth1");
+        Collection<? extends TreePath> treePaths = ctx.getMultiVariables().get("$var1$");
+        boolean factory = false;
+        for (TreePath treePath : treePaths) {
+            if (treePath.getLeaf() instanceof IdentifierTree) {
+                factory = true;
+            }
+        }
+        if (expr.equals("Executors") && methList.contains(method)) {
+            Fix fix = new FixImpl(ctx.getInfo(), ctx.getPath(), factory).toEditorFix();
+            if (factory) {
+                return ErrorDescriptionFactory.forName(ctx, ctx.getPath(), Bundle.ERR_ConvertToThreadPerTask(), fix);
+            } else {
+                return ErrorDescriptionFactory.forName(ctx, ctx.getPath(), Bundle.ERR_ConvertToVT(), fix);
+            }
+        }
+        return null;
+    }
+
+    private static final class FixImpl extends JavaFix {
+
+        boolean factory;
+
+        public FixImpl(CompilationInfo info, TreePath main, boolean factory) {
+            super(info, main);
+            this.factory = factory;
+        }
+
+        @Override
+        protected String getText() {
+            if (!factory) {
+                return Bundle.FIX_ConvertToVT();
+            } else {
+                return Bundle.ERR_ConvertToThreadPerTask();
+            }
+        }
+
+        @Override
+        protected void performRewrite(TransformationContext ctx) throws Exception {
+            if (!factory) {
+                WorkingCopy wc = ctx.getWorkingCopy();
```

Comment:
FWIW, there is:
`org.netbeans.spi.java.hints.JavaFixUtilities.rewriteFix` which allows to specify the re-write in a textual form. E.g.:
`$var = $expr2.newVirtualThreadPerTaskExecutor($var1$);` (or, better: `java.util.concurrent.Executors.newVirtualThreadPerTaskExecutor()`, if the first suggestion is adopted.)